exporter/prometheusremotewriteexporter/helper.go

```diff
@@ -5,30 +5,41 @@ package prometheusremotewriteexporter // import "github.com/open-telemetry/opent
 
 import (
 	"errors"
+	"math"
 	"sort"
 
 	"github.com/prometheus/prometheus/prompb"
 )
 
+// Track batch sizes sent to avoid over allocating huge buffers.
+// This helps in the case where large batches are sent to avoid allocating too much unused memory
+var nextTimeSeriesBufferSize int = math.MaxInt
+var nextMetricMetadataBufferSize int = math.MaxInt
+var nextRequestBufferSize int = 0
+
 // batchTimeSeries splits series into multiple batch write requests.
 func batchTimeSeries(tsMap map[string]*prompb.TimeSeries, maxBatchByteSize int, m []*prompb.MetricMetadata) ([]*prompb.WriteRequest, error) {
 	if len(tsMap) == 0 {
 		return nil, errors.New("invalid tsMap: cannot be empty map")
 	}
 
-	requests := make([]*prompb.WriteRequest, 0, len(tsMap)+len(m))
-	tsArray := make([]prompb.TimeSeries, 0, len(tsMap))
+	// Allocate a buffer size of at least 10, or twice the last # of requests we sent
```

Comment:
IMO these optimizations should be safe but we could put them behind a feature flag or config switch if there is any concern. please let me know what you prefer.

Reply:
I think it is fine to not use a feature flag here.